### ISSUE TITLE
Inference scripts

### ISSUE THREAD

**Issue by frankvp11** [NVIDIA/TensorRT#2216]
## Description
 Are there are any good scripts out there that can infer without needing detectron2 installed, as i'm running into issues installing detectron2 on python3.6 and the script that was put out in the /samples/python/detectron2 directory. 

## Environment

**TensorRT Version**: 8.2.1
**NVIDIA GPU**:  Jetson TX2
**NVIDIA Driver Version**: 10.2
**CUDA Version**: 10.2
**CUDNN Version**:  unknown
**Operating System**: Ubuntu 18.04
**Python Version (if applicable)**:  3.6.9
**Tensorflow Version (if applicable)**:  NA
**PyTorch Version (if applicable)**:  torch 1.10.0
**Baremetal or Container (if so, version)**: Baremetal right now - looking to move to container

## Relevant Files
The relevant files are in /samples/python/detectron2

## Steps To Reproduce

I am just trying to get by with python3.6.9 as I am unable to install detectron2 onto python3.6, so I am wondering if there are any scripts that don't need Detectron2 installed (or if someone knows how to get detectron2 onto python3.6) that work.

**Comment by zerollzeng** [comment]
@azhurkevich Can you help here ^ ^ many thanks!

**Comment by azhurkevich** [comment]
Do not understand how it is a TensorRT related issue, you won't be able to run /samples/python/detectron2 scripts without detectron2, if you don't want it, replace all detectron2 calls with particular values that are extracted in the process. You can use other samples for reference if you want ([1](https://github.com/NVIDIA/TensorRT/tree/main/samples/python/tensorflow_object_detection_api), [2](https://github.com/NVIDIA/TensorRT/tree/main/samples/python/efficientdet)).

**Comment by frankvp11** [comment]
Yeah thank you- these are exactly the type of scripts i'm looking for

**Comment by RajUpadhyay** [comment]
I know this is an old issue but use c++ or deepstream for inference when dealing with tensorrt on a jetson.
It gives way better performance.
I personally got upto 29 FPS with detectron2 int8 precision when I used deepstream.

**Comment by azhurkevich** [comment]
@RajUpadhyay thank you for running our sample and verifying that it works on Jetson. I wrote the sample but never verified it on a Jetson device myself)

**Comment by RajUpadhyay** [comment]
@azhurkevich thank you for the sample, it helped alot.
For inference it took some time but I was able to do it with c++ and later was able to use it with deepstream too.
I have already tried it with mask_rcnn and faster_rcnn (basically mrcnn haha, i just removed the mask part). Although I believe I need to wait a bit more to use keypoint models.
If anyone would want to try detectron2 with deepstream, they can refer [this](https://github.com/RajUpadhyay/Detectron2-Deepstream) git repo I made for fun. It is just a sample.

**Comment by azhurkevich** [comment]
@RajUpadhyay Thank you so much! I had a rough version of a keypoint network working for one of the projects, but did have time to push it publicly. It is definitely possible and doesn't require too much extra work. Doable in a couple days if you are familiar with sample's code structure. Glad you found it useful)

**Comment by RajUpadhyay** [comment]
OK I will give it a try, I did make an onnx but it only gave heatmaps and not the actual points.
Now that I know it is doable, I will try to do it one more time.
Thanks for letting me know! Very much appreciate it.

**Comment by azhurkevich** [comment]
Yep, sorry I wish I could help, but it takes time to commit + release + I need to revisit 1year+ old code that was not 100% finished with evaluation code. Please consider making a contribution eventually because a lot of people will find it useful. We can test it and merge.

One fact that I remember is that keypoint head is quite expensive. By default I think there are a lot of keypoints that are hardly used. I want to say default is like 80 or smth. Hence if you need less (which is most likely the case), before exporting to ONNX setting the number of keypoints to the amount that interests you will give a very significant boost.

**Comment by RajUpadhyay** [comment]
Ok wow thanks. 
If I do eventually manage to make an onnx for keypoints, I'll ping you.
Thanks again.

**Comment by ttyio** [comment]
closing issues with no activity for more than 3 weeks per our policy, thanks all!
